### PR TITLE
Added RandomForest.md #958

### LINKED ISSUE #958: [Feature Request] Random Forest Algorithm in ML
### Is your feature request related to a problem? Please describe.
Currently, decision tree-based models often suffer from overfitting or lack robustness in certain datasets. Using a single decision tree can be unreliable for high-dimensional or noisy data. A Random Forest algorithm would help mitigate these issues by aggregating multiple decision trees, leading to better predictive performance.

### Describe the solution you'd like
Implement a **Random Forest Algorithm**, which constructs a collection of decision trees (a forest) during training and outputs the mode of classes for classification or the average prediction for regression. This ensemble approach would improve model accuracy, reduce overfitting, and handle both categorical and continuous features more efficiently.

### Describe alternatives you've considered
I have considered boosting techniques or bagging individual decision trees. However, a Random Forest is preferable for handling noise and complex datasets without requiring excessive tuning.

### Additional context
Adding this feature would enhance the model's scalability and flexibility, making it suitable for a wide range of real-world applications, including classification and regression tasks in various domains like finance, healthcare, and e-commerce.

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by kartik1112**
### Description
This Feature adds a new ML algorithm Random Forest

Fixes #958 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/machine-learning/RandomForest.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+# Train model
+clf.fit(X_train, y_train)
+
+# Predict
+y_pred = clf.predict(X_test)
+
+# Evaluate
+accuracy = accuracy_score(y_test, y_pred)
+print(f"Accuracy: {accuracy:.2f}")
```
Comment (ajay-dhangar) [inline docs/machine-learning/RandomForest.md]: close your code example with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/machine-learning/RandomForest.md (hunk 1 above)
Comment (kartik1112) [reply docs/machine-learning/RandomForest.md]: done ✅

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**